lib/core/src/Cardano/Pool/Metadata.hs

```diff
@@ -0,0 +1,345 @@
+fetchStakePoolMetaZip :: String -> Handle -> IO (Either FetchError Integer)
+fetchStakePoolMetaZip url zipFile = tryJust handler $ do
+    req <- parseUrlThrow url
+    httpSink req $ \_response -> Conduit.mapM_ (BS.hPut zipFile)
+    hTell zipFile
+  where
+    handler = Just . FetchErrorDownload . displayException @HttpException
+
+-- | With the given zip file, extract and parse metadata for the given stake
+-- pools.
+getMetadataFromZip
+    :: Trace IO RegistryLogMsg
+    -> FilePath -- ^ Zip file path.
+    -> [PoolId] -- ^ Stake pools to extract.
+    -> IO [Maybe StakePoolMetadata]
+getMetadataFromZip tr zipFileName =
+    withArchive zipFileName . mapM (findStakePoolMeta tr . registryFile)
+
+-- | Try to read and parse a metadata JSON file within a 'ZipArchive', if it is
+-- present.
+--
+-- Any validation/parse error is unexpected because the data should already be
+-- validated. We shall simply log this occurrence and return 'Nothing'.
+--
+-- This may throw "Codec.Archive.Zip.EntrySelectorException" if there is an
+-- internal error and the pool id results in an invalid filename.
+findStakePoolMeta
+    :: Trace IO RegistryLogMsg -- ^ Logging.
+    -> FilePath -- ^ Filename within zip file to work on.
+    -> ZipArchive (Maybe StakePoolMetadata)
+findStakePoolMeta tr entry = do
+    trace $ MsgExtractFile entry
+    findArchiveFile entry >>= \case
+        Just sel -> do
+            json <- getEntry sel
+            case eitherDecodeStrict json of
+                Right meta -> do
+                    trace $ MsgExtractFileResult (Just (Right meta))
+                    pure $ Just meta
+                Left err -> do
+                    trace $ MsgExtractFileResult (Just (Left err))
+                    pure Nothing
+        Nothing -> do
+            trace $ MsgExtractFileResult Nothing
+            pure Nothing
+  where
+    trace = liftIO . registryLog tr
+
+-- | A GitHub repo zip file archive contains the files beneath a top level
+-- directory (which has a variable name). This function searches the archive
+-- entries for the one which corresponds to a JSON file for the pool ID.
+findArchiveFile :: FilePath -> ZipArchive (Maybe EntrySelector)
+findArchiveFile repoPath = find isRepoFile . Map.keys <$> getEntries
+  where
+    isRepoFile entry = dropTopLevel (unEntrySelector entry) == repoPath
+    dropTopLevel = drop 1 . dropWhile (not . isPathSeparator)
+
+-- | The path within the registry repo of metadata for a stake pool.
+registryFile :: PoolId -> FilePath
+registryFile (PoolId poolId) = "registry" </> B8.unpack poolId <.> "json"
+
+-- | The stake pool registry zipfile download URL for CF.
+cardanoFoundationRegistryZip :: String
+cardanoFoundationRegistryZip =
+    "https://github.com/" <> owner <> "/" <> repo <> "/archive/" <> archive
+  where
+    owner = "cardano-foundation"
+    repo = "incentivized-testnet-stakepool-registry"
+    archive = ref <> ".zip"
+    ref = "master"
```

Comment:
**Question:**

We have this hardcoded, but what if we'd like to test how the mechanism working on some other testnet, QA testnet for example?

Reply:
@piotr-iohk in order to keep the requirements and development simple with the story, we've chosen to not make this configurable _yet_. There are already a lot to do with this and allowing a command-line option makes the story longer by at least one or two days. 

The module is well tested at the "unit" level (though the unit test are actually firing up a web server serving zip files, so very much like it'd work with the foundation). 

Also, as far as I know, there is only one "official" registry hosted by the foundation and there are no plans to have different registry for the testnets.